plugin/core/registry.py

```diff
@@ -161,3 +168,143 @@ def run_async() -> None:
+SERVER_CAPABILITIES = {
+    "completionProvider": "Auto Complete",
+    "hoverProvider": "Hover",
+    "signatureHelpProvider": "Signature Help",
+    "declarationProvider": "Goto Declaration",
+    "definitionProvider": "Goto Definition",
+    "typeDefinitionProvider": "Goto Type Definition",
+    "implementationProvider": "Goto Implementation",
+    "referencesProvider": "Find References",
+    "documentHighlightProvider": "Highlights",
+    "documentSymbolProvider": "Goto Symbol",
+    "codeActionProvider": "Code Actions",
+    "codeLensProvider": "Code Lenses",
+    "documentLinkProvider": "Links",
+    "colorProvider": "Color Boxes",
+    "documentFormattingProvider": "Formatting",
+    "documentRangeFormattingProvider": "Range Formatting",
+    # "documentOnTypeFormattingProvider": "On Type Formatting",  # not supported by this client
+    "renameProvider": "Rename",
+    # "foldingRangeProvider": "Folding",  # not supported by this client
+    "executeCommandProvider": "Run Server Command",
+    "selectionRangeProvider": "Expand Selection",
+    # "linkedEditingRangeProvider": "Linked Editing",  # not supported by this client
+    # "callHierarchyProvider": "Call Hierarchy",  # not supported by this client
+    "semanticTokensProvider": "Semantic Highlighting",
+    # "typeHierarchyProvider": "Type Hierarchy",  # not supported by this client
+    "inlayHintProvider": "Inlay Hints",
+    # "diagnosticProvider": "Pull Diagnostics",  # not supported by this client
+    "workspaceSymbolProvider": "Goto Symbol in Project"
+}
+
+
+class LspToggleCapabilityCommand(LspWindowCommand):
+
+    last_toggled = None
+
+    def run(self, capability: str) -> None:
+        global toggled_capabilities
+        if capability not in SERVER_CAPABILITIES:
+            raise ValueError("Invalid capability name: {}".format(capability))
+        if capability in toggled_capabilities:
+            toggled_capabilities.remove(capability)
+            new_state = "on"
+        else:
+            toggled_capabilities.add(capability)
+            new_state = "off"
+        self.last_toggled = capability
+        if capability == "hoverProvider":
+            user_setting = sublime.load_settings("Preferences.sublime-settings").get("show_definitions")
+            for session in self.sessions():
+                for sv in session.session_views_async():
+                    if new_state == "on":
+                        sv.view.settings().set("show_definitions", False)
+                    else:
+                        sv.view.settings().set("show_definitions", user_setting)
+        elif capability == "codeLensProvider":
+            for session in self.sessions():
+                for sv in session.session_views_async():
+                    if new_state == "on":
+                        sv.start_code_lenses_async()
+                    else:
+                        sv.clear_all_code_lenses()
```

Comment:
I wonder if it would be cleaner to expose `override_capability` (or whatever is the appropriate name) on the `Session` instead since it already has `get_capability`/`has_capability`. Feels like it would have better parity there.

Reply:
Which then could also be useful for LSP-* plugins to use (see one potential use case in https://github.com/sublimelsp/LSP-css/issues/43).

While the plugin could trigger the command directly instead, it's not ideal as it's "fire and forget" without a return value so there is no feedback that would be good to have in a plugin.